test/unittest/middleware/ced/CedMiddlewareTests.cpp

```diff
@@ -0,0 +1,907 @@
+
+    /* Create
+     *      Id:         different
+     *      Publisher:  same
+     *      Topic:      non-existent
+     *      Expected:   FALSE
+     */
+    EXPECT_FALSE(middleware_.create_datawriter_by_ref(3, 0, "OtherTopic", associated_topic));
+}
+
+TEST_F(CedMiddlewareUnitTests, CreateDataWriterByXML)
+{
+    std::string participant_ref{"Participant"};
+    middleware_.create_participant_by_ref(0, 0, participant_ref);
+
+    std::string topic_ref_one{"TopicOne"};
+    std::string topic_ref_two{"TopicTwo"};
+    uint16_t topic_id_one = 1;
+    uint16_t topic_id_two = 2;
+    middleware_.create_topic_by_ref(topic_id_one, 0, topic_ref_one);
+    middleware_.create_topic_by_ref(topic_id_two, 0, topic_ref_two);
+
+    std::string publisher_xml{"Publisher"};
+    middleware_.create_publisher_by_xml(0, 0, publisher_xml);
+    middleware_.create_publisher_by_xml(1, 0, publisher_xml);
+
+    std::string datawriter_xml_one{topic_ref_one};
+    std::string datawriter_xml_two{topic_ref_two};
+    uint16_t associated_topic;
+
+    /* Create DataWriter by XML. */
+    EXPECT_TRUE(middleware_.create_datawriter_by_xml(0, 0, datawriter_xml_one, associated_topic));
+    EXPECT_EQ(associated_topic, topic_id_one);
+
+    /* Create
+     *      Id:         same
+     *      Publisher:  same
+     *      Topic:      same
+     *      Expected:   FALSE
+     */
+    EXPECT_FALSE(middleware_.create_datawriter_by_xml(0, 0, datawriter_xml_one, associated_topic));
+
+    /* Create
+     *      Id:         different
+     *      Publisher:  same
+     *      Topic:      same
+     *      Expected:   TRUE
+     */
+    EXPECT_TRUE(middleware_.create_datawriter_by_xml(1, 0, datawriter_xml_one, associated_topic));
+
+    /* Create
+     *      Id:         same
+     *      Publisher:  different
+     *      Topic:      same
+     *      Expected:   FALSE
+     */
+    EXPECT_FALSE(middleware_.create_datawriter_by_xml(0, 1, datawriter_xml_one, associated_topic));
+
+    /* Create
+     *      Id:         same
+     *      Publisher:  non-existent
+     *      Topic:      same
+     *      Expected:   FALSE
+     */
+    EXPECT_FALSE(middleware_.create_datawriter_by_xml(0, 2, datawriter_xml_one, associated_topic));
+
+    /* Create
+     *      Id:         same
+     *      Publisher:  same
+     *      Topic:      different
+     *      Expected:   FALSE
+     */
+    EXPECT_FALSE(middleware_.create_datawriter_by_xml(0, 0, datawriter_xml_two, associated_topic));
+
+    /* Create
+     *      Id:         same
+     *      Publisher:  same
+     *      Topic:      non-existent
+     *      Expected:   FALSE
+     */
+    EXPECT_FALSE(middleware_.create_datawriter_by_xml(0, 0, "OtherTopic", associated_topic));
+
+    /* Create
+     *      Id:         different
+     *      Publisher:  same
+     *      Topic:      different
+     *      Expected:   TRUE
+     */
+    EXPECT_TRUE(middleware_.create_datawriter_by_xml(2, 0, datawriter_xml_two, associated_topic));
```

Comment:
The id `2` and `3` (next line) has been created? Do you mean `1`?

Reply:
No, I use different ids to check the correct behaviour.